lib/src/luarule.cc

```diff
@@ -80,31 +79,40 @@ void LuaRule::code(const std::string& newCode)
     }
 }
 
+static std::string auditValue(const std::string& metric, double value)
+{
+    char svalue[32] = "";
+    if (!std::isnan(value)) {
+        snprintf(svalue, sizeof(svalue), "%0.2lf", value);
+        char* p = strstr(svalue, ".00");
+        if (p) { *p = 0; } // remove .00 decimals
+    }
+    // <metricName>=<value> format
+    return metric.substr(0, metric.find("@")) + "=" + std::string{svalue};
+}
+
 int LuaRule::evaluate(const MetricList& metricList, PureAlert& pureAlert)
 {
     log_debug("LuaRule::evaluate %s", _name.c_str());
     int res = 0;
 
-    std::vector<double>      values;
-    std::vector<std::string> auditValues;
-    int                      index = 0;
+    std::string auditValues;
+
+    std::vector<double> values;
+    int index = 0;
     for (const auto& metric : _metrics) {
         double value = metricList.find(metric);
+
```

Comment:
We lost the information NaN which is different of empty value

Reply:
metricList.find() returns nan if the metric is not monitored (shm). 
Here, NaN does not means not a number   :(
I prefer to display an empty string here (more neutral)